packages/crayons-core/src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  Component,
+  Prop,
+  h,
+  Host,
+  Event,
+  EventEmitter,
+  Method,
+  State,
+} from '@stencil/core';
+@Component({
+  tag: 'fw-pagination',
+  styleUrl: 'pagination.scss',
+  shadow: true,
+})
+export class Pagination {
+  @State() end;
+  @State() start;
+
+  /**
+   * The current page number.
+   */
+  @Prop() page = 1;
+  /**
+   * The total number of records.
+   */
+  @Prop({ mutable: true }) totalRecords: number;
+  /**
+   *The number of records to be shown per page. Defaults to 10.
+   */
+  @Prop() recordsPerPage = 10;
+  /**
+   * Triggered when either previous or next button is clicked.
+   */
+  @Event() fwChange: EventEmitter;
+
+  @Method()
+  async previous() {
+    this.goToPrevious();
+  }
+
+  @Method()
+  async next() {
+    this.goToNext();
+  }
+
+  componentWillLoad() {
+    // If both totalRecords and page are passed in but if the start record of current page comes out to be
+    // greater than total number of records then assign start to the last page's first record number.
+    this.start = this.totalRecords
```

Comment:
rather than do this let's make sure `this.page` is pointing to the last offset if it's pointing to an out of bounds pointer.

Reply:
Have made the necessary changes in latest commit by making `total` as mandatory param